main/guides/coreeval/local-testnet.md

```diff
@@ -1,45 +1,189 @@
+     since the core-eval scope includes exports of `@endo/far`.
+   - `bundleID = ...` is replaced using updated/cached bundle hash
+   - `main` is appended as the script completion value
+   - the `permit` export is emitted to a `.json` file
 
-    For example, to deploy the PSM proposal referenced in the previous topics, run the following:
+See [rollup.config.mjs](https://github.com/Agoric/dapp-agoric-basics/blob/dc-vote-fee/contract/rollup.config.mjs) for details on using the
+`moduleToScript`, `configureBundleID`, and `emitPermit` rollup plug-ins.
 
-    ```
-	./agd --chain-id=agoriclocal --title=<Insert your own title> --description=<Insert your description> --home=<PATH to your project folder>/_agstate/keys --keyring-backend=test --from=provision tx gov submit-proposal swingset-core-eval ../packages/inter-protocol/test/psm/gov-add-psm-permit.json ../packages/inter-protocol/test/psm/gov-add-psm.js
-    ```
+Running it looks something like this:
+
+```console
+dapp-agoric-basics/contract$ yarn build:deployer
+yarn run v1.22.19
+$ rollup -c rollup.config.mjs
+
+./src/sell-concert-tickets.proposal.js → bundles/deploy-sell-concert-tickets.js...
+bundles add: sell-concert-tickets from ./src/sell-concert-tickets.contract.js
+bundles bundled 141 files in bundle-sell-concert-tickets.js at 2024-03-09T16:22:08.531Z
+created bundles/deploy-sell-concert-tickets.js in 2.4s
+...
+
+dapp-agoric-basics/contract$ cd bundles
+dapp-agoric-basics/contract/bundles$ ls deploy-sell* bundle-sell*
+bundle-sell-concert-tickets.js          deploy-sell-concert-tickets-permit.json deploy-sell-concert-tickets.js
+```
+
+## Sending Transactions to Deploy a Contract
+
+To submit transactions to get the contract and core eval script and permit on chain, you can use a web UI or CLI tools.
+
+### Ensuring you have enough IST to install a bundle
+
+Installing a contract costs 0.02IST per kilobyte.
+
+On a local chain, use `yarn docker:make mint100` to get 100 IST, which should be enough. _`mint100`  works by sending some ATOM from a "whale" account and using the ATOM to open a vault._
+
+On devnet, use the [devnet faucet](https://devnet.faucet.agoric.net/) to get 25IST. You might need to use  2 or 3 times.
+
+### Installing, submitting with **cosgov** Web UI
+
+The Cosmos Proposal Builder at [cosgov.org](https://cosgov.org) is one way to prepare the transactions. It supports the local chain as well as **devnet** etc.
+
+You'll need a `.json` formatted bundle:
+
+```console
+dapp-agoric-basics/contract$ yarn bundle-source --cache-json bundles/ src/sell-concert-tickets.contract.js sell-concert-tickets
+bundles/ add: sell-concert-tickets from src/sell-concert-tickets.contract.js
+bundles/ bundled 141 files in bundle-sell-concert-tickets.json at 2024-03-09T20:48:43.784Z
+```
+
+<img src='../assets/cosgov-install-bundle.png' alt='Install Bundle screenshot' width="600" />
+
+Then drop your permit and core eval script on the **CoreEval Proposal** tab:
```

Comment:
_nitpick_

You probably meant to add another newline before the <img> tab below?